src/rascal/representations/calculator_spherical_expansion.hh

```diff
@@ -1716,35 +1729,34 @@ namespace rascal {
                                                                      neigh);
         double f_c{cutoff_function->f_c(dist)};
         auto coefficients_center_by_type{coefficients_center[neigh_type]};
-
-        // compute the coefficients
         size_t l_block_idx{0};
-        for (size_t angular_l{0}; angular_l < this->max_angular + 1;
-             ++angular_l) {
-          size_t l_block_size{2 * angular_l + 1};
-          c_ij_nlm.block(0, l_block_idx, max_radial, l_block_size) =
-              neighbour_contribution.col(angular_l) *
-              harmonics.segment(l_block_idx, l_block_size);
-          l_block_idx += l_block_size;
-        }
-        c_ij_nlm *= f_c;
-        coefficients_center_by_type += c_ij_nlm;
-
-        // half list branch for c^{ji} terms using
-        // c^{ij}_{nlm} = (-1)^l c^{ji}_{nlm}.
-        if (IsHalfNL) {
-          if (not manager->is_center_atom(atom_j)) {
-            std::stringstream err_str{};
-            err_str << "Half neighbor list should only be used when all the "
-                    << "atoms inside the unit cell are centers, i.e. "
-                    << "center_atoms_mask should not mask atoms.";
-            throw std::runtime_error(err_str.str());
+        // compute the coefficients
+        // complications here because the infrastructure for swaping ij is not
+        // here yet.
+        // TODO(felix) make swaping ij possible and implement the
+        // proper half for gradients so the whole computation can be skipped.
+        if (not are_some_centers_masked and atom_i_tag > atom_j_tag) {
+          // continue;
+        } else {
+          for (size_t angular_l{0}; angular_l < this->max_angular + 1;
+               ++angular_l) {
+            size_t l_block_size{2 * angular_l + 1};
+            c_ij_nlm.block(0, l_block_idx, max_radial, l_block_size) =
+                neighbour_contribution.col(angular_l) *
+                harmonics.segment(l_block_idx, l_block_size);
+            l_block_idx += l_block_size;
           }
-          if (is_center_atom) {
+          c_ij_nlm *= f_c;
+          coefficients_center_by_type += c_ij_nlm;
+          // when j == i there is nothing to add to ji
+          if (not are_some_centers_masked and
+              (atom_j_tag != atom_i_tag)) {  // NOLINT
+            // half list branch for c^{ji} terms using
+            // c^{ij}_{nlm} = (-1)^l c^{ji}_{nlm}.
             auto & coefficients_neigh{expansions_coefficients[atom_j]};
             auto coefficients_neigh_by_type{coefficients_neigh[center_type]};
             l_block_idx = 0;
-            double parity{1.};
+            double parity{1.};  // account for (-1)^l
```

Comment:
The comment isn't very helpful, since it just repeats the code. Either take it out or explain that it comes from swapping i and j (and thereby negating `\hat{r}` as the argument to the spherical harmonics)

Reply:
it does not repeat the code to me since parity only holds for (-1)^l because there is a 
```parity *= -1.;``` 
later in the loop.

Reply:
Ok, then maybe just call the `(-1)^l` term in the comment 4 lines up a "parity term" explicitly, so that there's the link to the variable name below and you don't have to explain it with an extra comment.